### ISSUE TITLE
Getting cargo errors when running `melos build` 

### ISSUE THREAD

**Issue by achou11** [p2panda/meli#53]
Attempting to walk through the installation docs and when I run `melos build`, I get the following error from Cargo:

```sh
melos run build
  └> bash scripts/build.sh
     └> RUNNING

◆ Install Rust toolchain dependencies

    Updating crates.io index

     Ignored package `cargo-ndk v3.4.0` is already installed, use --force to override

    Updating crates.io index

  Installing cargo-expand v1.0.79

    Updating crates.io index

error: failed to compile `cargo-expand v1.0.79`, intermediate artifacts can be found at `/var/folders/3d/jpg9p2yn0t918hwkw50yrfxh0000gn/T/cargo-installjE8Dn7`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

Caused by:
  package `clap_derive v4.5.0` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.73.0
  Try re-running cargo install with `--locked`
```

I had run this command once before and I thought it worked, although I think I was mistaken so I ran it again, encountering the error above.

---

Environment info:

OS: macOS 14.2.1 (Sonoma)
Cargo: 1.73.0 (9c4383fb5 2023-08-26)
Melos: 3.1.1

**Comment by achou11** [comment]
very noobish when it comes to Rust + Cargo, but looks like `cargo-expand` specifies a dependency on clap v4 and the [latest version of clap v4 requires rustc 1.74.0](https://github.com/clap-rs/clap/blob/8a7a13a5618cfdc4ff328624a5266e7b4d88649a/Cargo.toml#L17). Attempted to use `cargo install cargo-expand --ignore-rust-version` instead but ran into compilation errors (probably because it needs features from the specified minimum version)

**Comment by adzialocha** [comment]
> very noobish when it comes to Rust + Cargo, but looks like `cargo-expand` specifies a dependency on clap v4 and the [latest version of clap v4 requires rustc 1.74.0](https://github.com/clap-rs/clap/blob/8a7a13a5618cfdc4ff328624a5266e7b4d88649a/Cargo.toml#L17). Attempted to use `cargo install cargo-expand --ignore-rust-version` instead but ran into compilation errors (probably because it needs features from the specified minimum version)

Right, it looks like that's the error: 

`package `clap_derive v4.5.0` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.73.0`

Does it work now?

**Comment by achou11** [comment]
> Does it work now?

trying with 1.74.1 and looks like things are building as expected (although taking a long time, probably also expected 😄 ). was a little hesitant to use a different version of Rust than what the docs say since I'm not sure how sensitive the project is when it comes to Rust version, but hopefully it's fine!

**Comment by achou11** [comment]
got this error now:

```sh
error: could not find native static library `clang_rt.builtins-x86_64-android`, perhaps an -L flag is missing?


error: could not compile `p2panda` (lib) due to previous error

note: If the build failed due to a missing target, you can run this command:
note: 
note:     rustup target install x86_64-linux-android
```

**Comment by achou11** [comment]
Ah my guess is that it's due to the hardcoding of the linux path here:

https://github.com/p2panda/meli/blob/c4017b1c3c2e184f55c125917ba4354ac9d6ba88/packages/p2panda/native/build.rs#L9

going to use `darwin-x86_64` and see what happens...

**Comment by achou11** [comment]
> going to use darwin-x86_64 and see what happens...

looks like this fixed it 👍

**Comment by achou11** [comment]
Solved with #54 and #55
